src/main/kotlin/com/zepben/evolve/services/network/translator/NetworkCimToProto.kt

```diff
@@ -536,6 +545,20 @@ fun toPb(cim: ConnectivityNode, pb: PBConnectivityNode.Builder): PBConnectivityN
 fun toPb(cim: ConnectivityNodeContainer, pb: PBConnectivityNodeContainer.Builder): PBConnectivityNodeContainer.Builder =
     pb.apply { toPb(cim, psrBuilder) }
 
+fun toPb(cim: Curve, pb: PBCurve.Builder): PBCurve.Builder =
+    pb.apply {
+        cim.data.forEachIndexed { i, data -> addCurveDataBuilder(i).apply { toPb(data, this) } }
+        toPb(cim, ioBuilder)
+    }
+
+fun toPb(cim: CurveData, pb: PBCurveData.Builder): PBCurveData.Builder =
+    pb.apply {
+        xvalue = cim.xValue
```

Comment:
todo: `xvalue` should have been `xValue` in the proto spec. This can be updated when you change the ratedU value to an int.

Reply:
This has been updated in https://github.com/zepben/ewb-grpc/tree/DEV-751-machines-and-more-tweaks, but is awaiting changes in the CI to allow it to be deployed/merged

Reply:
fixed up